tools/istio-iptables/main.go

```diff
@@ -0,0 +1,538 @@
+		// Jump to the ISTIO_OUTPUT chain from OUTPUT chain for all tcp traffic.
+		ip6tables("-t", "nat", "-A", "OUTPUT", "-p", "tcp", "-j", "ISTIO_OUTPUT").RunOrFail()
+		// Redirect app calls to back itself via Envoy when using the service VIP or endpoint
+		// address, e.g. appN => Envoy (client) => Envoy (server) => appN.
+		ip6tables("-t", "nat", "-A", "ISTIO_OUTPUT", "-o", "lo", "!", "-d", "::1/128", "-j", "ISTIO_REDIRECT").RunOrFail()
+
+		for _, uid := range split(proxyUID) {
+			// Avoid infinite loops. Don't redirect Envoy traffic directly back to
+			// Envoy for non-loopback traffic.
+			ip6tables("-t", "nat", "-A", "ISTIO_OUTPUT", "-m", "owner", "--uid-owner", uid, "-j", "RETURN").RunOrFail()
+		}
+
+		for _, gid := range split(proxyGID) {
+			// Avoid infinite loops. Don't redirect Envoy traffic directly back to
+			// Envoy for non-loopback traffic.
+			ip6tables("-t", "nat", "-A", "ISTIO_OUTPUT", "-m", "owner", "--gid-owner", gid, "-j", "RETURN").RunOrFail()
+		}
+		// Skip redirection for Envoy-aware applications and
+		// container-to-container traffic both of which explicitly use
+		// localhost.
+		ip6tables("-t", "nat", "-A", "ISTIO_OUTPUT", "-d", "::1/128", "-j", "RETURN").RunOrFail()
+		// Apply outbound IPv6 exclusions. Must be applied before inclusions.
+		for _, cidr := range ipv6RangesExclude.IPNets {
+			ip6tables("-t", "nat", "-A", "ISTIO_OUTPUT", "-d", cidr.String(), "-j", "RETURN").RunOrFail()
+		}
+		// Apply outbound IPv6 inclusions.
+		if ipv6RangesInclude.IsWildcard {
+			// Wildcard specified. Redirect all remaining outbound traffic to Envoy.
+			ip6tables("-t", "nat", "-A", "ISTIO_OUTPUT", "-j", "ISTIO_REDIRECT").RunOrFail()
+			for _, internalInterface := range split(kubevirtInterfaces) {
+				ip6tables("-t", "nat", "-I", "PREROUTING", "1", "-i", internalInterface, "-j", "RETURN").RunOrFail()
+			}
+		} else if len(ipv6RangesInclude.IPNets) > 0 {
+			// User has specified a non-empty list of cidrs to be redirected to Envoy.
+			for _, cidr := range ipv6RangesInclude.IPNets {
+				for _, internalInterface := range split(kubevirtInterfaces) {
+					ip6tables("-t", "nat", "-I", "PREROUTING", "1", "-i", internalInterface, "-d", cidr.String(), "-j", "ISTIO_REDIRECT").RunOrFail()
+				}
+				ip6tables("-t", "nat", "-A", "ISTIO_OUTPUT", "-d", cidr.String(), "-j", "ISTIO_REDIRECT").RunOrFail()
+			}
+			// All other traffic is not redirected.
+			ip6tables("-t", "nat", "-A", "ISTIO_OUTPUT", "-j", "RETURN").RunOrFail()
+		}
+	} else {
+		// Drop all inbound traffic except established connections.
+		ip6tables("-F", "INPUT").RunQuietlyAndIgnore()
+		ip6tables("-A", "INPUT", "-m", "state", "--state", "ESTABLISHED", "-j", "ACCEPT").RunQuietlyAndIgnore()
+		ip6tables("-A", "INPUT", "-i", "lo", "-d", "::1", "-j", "ACCEPT").RunQuietlyAndIgnore()
+		ip6tables("-A", "INPUT", "-j", "REJECT").RunQuietlyAndIgnore()
+	}
+}
+
+func main() {
+	// Emulate slow startup of bash
+	time.Sleep(time.Second * 5)
```

Comment:
Pretty sure it doesn't take 5 seconds, and it doesn't need to be emulated.

Reply:
This was just a try to fix the problems, we encountered using this go binary. It was not successful. Therefore, we will remove this.